app/src/main/java/com/example/peterchu/watplanner/Views/CourseDetailActivity.java

```diff
@@ -0,0 +1,117 @@
+                    CourseDetailFragment.ARG_COURSE_ID, -1));
+            CollapsingToolbarLayout appBarLayout = (CollapsingToolbarLayout) this.findViewById(
+                    R.id.toolbar_layout);
+            if (appBarLayout != null) {
+                appBarLayout.setTitle(course.getName());
+            }
+//            final CourseSchedule courseSchedule = dbHandler.getCourseSchedule(course.getSubject(),
+//                    course.getNumber());
+            ApiInterface apiService =
+                    ApiClient.getClient().create(ApiInterface.class);
+            Call<CourseDetailsResponse> call = apiService.getCourseDetails(course.getSubject(),
+                    course.getNumber(), Constants.API_KEY);
+
+            call.enqueue(new Callback<CourseDetailsResponse>() {
+                @Override
+                public void onResponse(Call<CourseDetailsResponse> call, Response<CourseDetailsResponse> response) {
+                    Log.d("MainActivity", response.toString());
+                    CourseDetails courseDetails = response.body().getData();
+                    // Create the detail fragment and add it to the activity
+                    // using a fragment transaction.
+                    Bundle arguments = new Bundle();
+                    arguments.putInt(CourseDetailFragment.ARG_COURSE_ID,
+                            getIntent().getIntExtra(CourseDetailFragment.ARG_COURSE_ID, -1));
+                    CourseDetailFragment fragment = new CourseDetailFragment();
+                    fragment.setCourseDetails(courseDetails);
+//                    fragment.setCourseSchedule(courseSchedule);
+                    fragment.setArguments(arguments);
+                    getSupportFragmentManager().beginTransaction()
+                            .add(R.id.course_detail_container, fragment)
+                    .commit();
+
+                }
+
+                @Override
+                public void onFailure(Call<CourseDetailsResponse> call, Throwable t) {
+                    // Log error here since request failed
+                    Log.e("CourseDetailFragment", t.toString());
+                }
+            });
+
+            // Show the Up button in the action bar.
+            ActionBar actionBar = getSupportActionBar();
+            if (actionBar != null) {
+                actionBar.setDisplayHomeAsUpEnabled(true);
+            }
+        }
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        int id = item.getItemId();
+        if (id == android.R.id.home) {
+            // This ID represents the Home or Up button. In the case of this
+            // activity, the Up button is shown. For
+            // more details, see the Navigation pattern on Android Design:
+            //
+            // http://developer.android.com/design/patterns/navigation.html#up-vs-back
+            //
+            navigateUpTo(new Intent(this, MainActivity.class));
```

Comment:
Should we implement OS back button & app up arrow button to just `finish()` the activity? This way, if the user comes from the search screen or the home screen (just as an example), the user can return to their previous page rather than always going back to`MainActivity`